modules/system/defaults-write.nix

```diff
@@ -109,8 +109,11 @@ in
         ${concatStringsSep "\n" CustomUserPreferences}
 
         ${optionalString (length dock > 0) ''
-          echo >&2 "restarting Dock..."
-          killall Dock
+          # Only restart Dock if current user is logged in
+          if pgrep -xu $UID Dock; then
```

Comment:
Shouldn’t need this with the `|| true`.

Reply:
I would rather leave it in as it makes the intent of this code more clear (the prerequisite of restarting the Dock is when it is running as the current user)

Reply:
Alright; it’ll be a little awkward in a post‐user‐activation world but we can cross that bridge when we come to it.